deployment/ccip/changeset/deployer_group.go

```diff
@@ -41,12 +64,21 @@ type DeployerGroup struct {
 //	state.Chains[selector].RMNRemote.Curse()
 //	# Execute the transaction or create the proposal
 //	deployerGroup.Enact("Curse RMNRemote")
-func NewDeployerGroup(e deployment.Environment, state CCIPOnChainState, mcmConfig *MCMSConfig) *DeployerGroup {
+func NewDeployerGroup(e deployment.Environment, state CCIPOnChainState, deploymentContext *DeploymentContext, mcmConfig *MCMSConfig) *DeployerGroup {
```

Comment:
NewDeployerGroup should enforce an empty initial context. I like the idea of this builder-style API:

```
deployerGroup := NewDeployerGroup(e, state, mcmsConfig).WithDeploymentContext("set RMNHome dynamic config")

...

// Context update
deployerGroup = deployerGroup.WithDeploymentContext("do something else")
```